compiler/rustc_lint/src/empty_iterator.rs

```diff
@@ -0,0 +1,76 @@
+use crate::{EarlyContext, EarlyLintPass, LintContext};
+
+use rustc_ast::{Expr, ExprKind, Lit, LitKind};
+use rustc_errors::fluent;
+
+declare_lint! {
+    /// The `empty_for_iterator` lint checks for empty `for` iterator.
+    ///
+    /// ### Example
+    ///
+    /// ```rust
+    /// for i in 10..0 { /* ... */ }
+    /// ```
+    ///
+    /// {{produces}}
+    ///
+    /// ### Explanation
+    ///
+    /// It is usually a mistake to have a statement that has no effect.
+    pub EMPTY_FOR_ITERATOR,
+    Warn,
+    "empty for iterator"
```

Comment:
The lint name is somewhat confusing to me, it reads as `empty (for iterator)`, but for-iterator is not a thing. Maybe `for_loop_over_empty_iterator` (to be similar with [`for_loop_over_fallibles`](https://github.com/rust-lang/rust/pull/99696)) or `for_in_empty_iterator` would be better?

Reply:
I also suggested `empty_iter_ranges` given its scope of detecting `(10..0).rev()`